rgw/admin/link.go

```diff
@@ -0,0 +1,44 @@
+//go:build ceph_preview
+// +build ceph_preview
+
+package admin
+
+import (
+	"context"
+	"net/http"
+)
+
+// BucketLinkInput the bucket link/unlink input parameters
+type BucketLinkInput struct {
+	Bucket   string `url:"bucket" json:"bucket"`
+	BucketID string `url:"bucket-id" json:"bucket_id"`
+	UID      string `url:"uid" json:"uid"`
+}
+
+// UnlinkBucket unlink a bucket from a specified user
+// Primarily useful for changing bucket ownership.
+//  PREVIEW
+func (api *API) UnlinkBucket(ctx context.Context, link BucketLinkInput) error {
+	if link.UID == "" {
+		return errMissingUserID
+	}
+	if link.Bucket == "" {
+		return errMissingBucket
+	}
+	_, err := api.call(ctx, http.MethodPost, "/bucket", valueToURLParams(link))
+	return err
+}
+
+// LinkBucket will link a bucket to a specified user
+// unlinking the bucket from any previous user
```

Comment:
@elrondwong : Sorry for the dump question, I don't understand completely about this comment, if you link with New user, then previous user will automatically unlinked??

Reply:
yes, I have try, but the unlink api makes no sense.

Reply:
[official docs](https://docs.ceph.com/en/pacific/radosgw/adminops/#unlink-bucket) show that.

Reply:
> yes, I have tried, but the unlink api makes no sense.

AFAIR user cannot be removed until all the buckets attached to it are removed, so if u want to keep data then just unlink the user for the time being and link the bucket with a new user whenever it creates. Now a day

Reply:
Yes! I just tested it and it's still today.
Thank you both for great reviews  and for guidance!!! 🥳